### PR TITLE
refactor(tests): replace custom BuiltinData format with UPLC text + Plutus JSON

### LINKED ISSUE #148: Use standard Plutus Data JSON encoding for builtin_data values in cape-tests.json
## Problem

The `cape-tests.json` files use a custom compact notation for `builtin_data` values:

```json
{
  "deposited_escrow_datum": {
    "type": "builtin_data",
    "value": "0(0() 1000)"
  }
}
```

This notation (`0(fields...)` for constructors, `#hex` for bytestrings, `[a b]` for lists, `{k:v}` for maps, plain integers) is not a standard format and is not formally specified anywhere in the CAPE documentation. Every compiler submission must implement a bespoke parser for it, which is unnecessary friction.

## Suggestion

Replace the custom notation with one of the two existing standard Plutus Data representations that every Cardano toolchain already supports:

**Option A: Plutus JSON schema** (used by `cardano-cli`, Aiken, Plu-ts, etc.)

```json
{
  "deposited_escrow_datum": {
    "type": "builtin_data",
    "value": {"constructor": 0, "fields": [{"constructor": 0, "fields": []}, {"int": 1000}]}
  }
}
```

This is the `ScriptDataJsonDetailedSchema` from `cardano-cli`. It is widely adopted and well-documented in the [Cardano CLI docs](https://github.com/IntersectMBO/cardano-cli).

**Option B: UPLC text Data syntax** (from the Plutus spec)

```json
{
  "deposited_escrow_datum": {
    "type": "builtin_data",
    "value": "Constr 0 [Constr 0 [], I 1000]"
  }
}
```

This is the Data syntax used inside `(con data ...)` in UPLC text format. Every UPLC parser already handles it.

## Benefits

- Every Cardano toolchain already has a parser for at least one of these formats
- No custom parser needed per submission -- reduces onboarding effort for new compilers
- The format is formally specified (Plutus spec for Option B, CIP-related for Option A)
- Less room for ambiguity (e.g., is whitespace significant in the current format? How are nested maps handled?)

## Notes

The current format works fine and is reasonably readable. This is a quality-of-life suggestion to lower the barrier for new compiler submissions, not a blocking issue.

### PR DESCRIPTION
**Pull request by Unisay**
Closes #148.

## Summary

Replaces the bespoke compact BuiltinData notation (`0(#cafe 1000)`, `[1 2]`, `{1:2}`, `#deadbeef`) used in `cape-tests.json` with two **standard, widely-tooled** formats. Dispatch is by JSON type of the `value` field — no extra discriminator needed:

| JSON shape | Format | Example |
| --- | --- | --- |
| `string` | UPLC text Data syntax (Plutus Core spec) | `"Constr 0 [I 1000, B #cafe]"` |
| `object` | Plutus JSON detailed schema (cardano-cli) | `{"constructor": 0, "fields": [{"int": 1000}]}` |

`"@name"` reference shorthand is unchanged.

The custom parser at `lib/PlutusCore/Data/Compact/{Parser,Printer}.hs` is removed entirely; schema bumped 1.0.0 → 2.0.0.

## Why

Per @nau in #148: every compiler submission previously had to ship a parser for our bespoke format. Both replacement formats are already parsable out-of-the-box by every Cardano toolchain (UPLC text via any UPLC parser; Plutus JSON via cardano-cli-compatible JSON libraries). One less hand-rolled parser to maintain on our side too.

## Migration of existing cape-tests.json

Per the choice discussed for #148:

- **5 scenarios with Scalus submissions** → **Plutus JSON object form** (the format @nau preferred in the issue, native fit for Scala/JVM tooling): `factorial`, `factorial_naive_recursion`, `fibonacci`, `fibonacci_naive_recursion`, `htlc`.
- **3 remaining scenarios** → **UPLC text Data syntax** (closest readable analogue of the old compact form): `ecd`, `two_party_escrow`, `linear_vesting`.

Authors of new tests can mix both forms freely — pick whichever is most readable per slot.

## Implementation

- New modules `lib/Cape/Data/{UplcText,PlutusJson,Decode}.hs`. `UplcText` wraps the upstream `PlutusCore.Parser.conData`; `PlutusJson` implements the cardano-cli-style schema directly (~50 lines, no `cardano-api` dep).
- `Cape.Tests` exposes a new `ResolvedInput` ADT (`ResolvedUplc Text` / `ResolvedBuiltinData V3.BuiltinData`) and routes every BuiltinData slot — test inputs, redeemers, datums, output datums, asset bytes — through one `decodeBuiltinDataValue` helper that handles `@reference` resolution then delegates to the dispatch.
- As a side benefit: the previous `Data → compact text → Data` round-trip in `resolveScriptContextInput` is eliminated; `measure-app/Main.hs` consumes `ResolvedInput` directly.
- Old `lib/PlutusCore/Data/Compact/` modules deleted; corresponding specs replaced with `Cape.Data.UplcTextSpec` / `Cape.Data.PlutusJsonSpec` (round-trip + parser edge cases).
- ADR-0005 marked `superseded by 0006`; new [ADR-0006](doc/adr/0006-standard-builtindata-formats-with-json-type-dispatch.md) documents the rationale.

## Test plan

- [x] `cabal test cape-tests` — 241/241 pass.
- [x] `cape submission verify --all` — every existing submission across all 7 scenarios passes (covers Plinth submissions on UPLC-text-form scenarios and Scalus submissions on JSON-form scenarios).
- [x] CI green.

### CODE REVIEW

File: lib/Cape/Data/PlutusJson.hs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        | KeyMap.member "constructor" o -> do
+            ix <- o Json..: "constructor"
+            fields <- o Json..: "fields"
+            Constr ix <$> traverse parseData fields
+        | otherwise ->
+            fail $
+              "Plutus JSON Data object must contain one of "
+                <> "'int', 'bytes', 'list', 'map', 'constructor'; got keys: "
+                <> show (KeyMap.keys o)
+      other -> AesonTypes.typeMismatch "Plutus JSON Data object" other
```
[the remark this answers is not in the document]
Comment (Unisay) [reply lib/Cape/Data/PlutusJson.hs]: Fixed in a9729d3 — switched from `KeyMap.member` short-circuit to exact key-set matching (`{int}`, `{bytes}`, `{list}`, `{map}`, `{constructor, fields}`). Anything else (extra keys, multiple discriminators, partial constructor) now fails parse with a specific error message. Added regression tests in `PlutusJsonSpec` for each rejection class.

File: test/Cape/TestsSpec.hs [hunk 2]
```diff
@@ -7,404 +7,197 @@ import Data.Aeson qualified as Json
...
-        case parseBuiltinDataText result of
-          Left parseErr ->
-            expectationFailure $
-              "Failed to parse minimal ScriptContext: " <> show parseErr
-          Right _ -> pass
+        result <- resolveTestInput "" (mkSuite Nothing) testInput
+        case result of
+          ResolvedUplc t -> t `shouldBe` uplcProgram
+          ResolvedBuiltinData _ ->
+            expectationFailure "expected ResolvedUplc"
```
[the remark this answers is not in the document]
Comment (Unisay) [reply test/Cape/TestsSpec.hs:200]: Fixed in a9729d3 — changed to a term-only example (`(con integer 42)`), aligned with how `applyInputsToProgram` wraps UPLC inputs downstream. Renamed the binding from `uplcProgram` to `uplcTerm` for clarity and added a comment explaining the convention.
